corehq/apps/aggregate_ucrs/tests/data/table_definitions/monthly_aggregate_definition.yml

```diff
@@ -0,0 +1,52 @@
+domain: agg
+table_id: monthy_aggregate_pregnancies
+display_name: Pregnancy Data Per Month
+primary_table:
+  data_source_id: 550c3cd432d931387e75e8506b5caf9e  # the data source ID of e.g. the 'pregnancy_cases' table
+  key_column: doc_id
+  columns:
+    # this should generate SQL that looks something like:
+    # insert into pregnancy_cases... values... (
+    #   select name as name, lmp as pregnancy_start_date, 1 as open_in_month,
+    #   lmp < :month_end and (delivery_date is null or delivery_date >= :month_start) as pregnant_in_month
+    # )
+    - column_id: name
+      # todo: unclear where (if anywhere) we want to add dimension info. this is one possible place it could go
+      # is_dimension: false
+      type: reference
+      config_params:
+        referenced_column: name
+    - column_id: pregnancy_start_date
+      type: reference
+      config_params:
+        referenced_column: lmp
+    - column_id: open_in_month
+      type: constant
+      config_params:
+        constant: 1
+    - column_id: pregnant_in_month
+      type: sql_statement
+      config_params:
+        # sql statements will only be able to access the current row
+        datatype: small_integer
+        statement: CASE WHEN lmp < :month_end and (delivery_date is null or delivery_date >= :month_start) THEN 1 ELSE 0 END
+        statement_params:
+          # these '_agg' things are magic variables that let you reference the aggregation window's start/end conditions
+          month_start: :agg_window_start
+          month_end: :agg_window_end
+time_aggregation:
+  unit: month  # unit additionally determines the datatype and the column name used
+  # conditions for inclusion of a particular row on a particular month
+  # start and end conditions must (currently) be based off the primary data source table
+  start_column: opened_date  # if value is not set, defaults to excluding the row entirely
+  end_column: closed_date  # if value is not set, defaults to excluding the row beyond the current date
+secondary_tables:
+  # this is a list, though in this example there is only one.
+  - data_source_id: d824a4864ecb421fb3be8bf8173a05d7
+    key_column: form.case.@case_id  # how rows will get joined from primary to secondary tables
+    aggregation_column: received_on  # determines the aggregation window that each secondary row falls in
```

Comment:
I find this name a bit misleading since it seems like it would be column that get's aggregated. What about `window_column` or `filter_column`?

Reply:
good call. how about `time_window_column`? I had intended to add the `time_` prefix already and like `window` better than `aggregation` or `filter`

Reply:
`time_window_column` is good